packages/coding-standard/src/Rules/ForbiddenProtectedPropertyRule.php

```diff
@@ -37,6 +69,96 @@ public function process(Node $node, Scope $scope): array
             return [];
         }
 
+        if ($this->isInsideAbstractClassAndPassedAsDependencyViaConstructor($node)) {
+            return [];
+        }
+
+        if ($this->isStaticAndContainerOrKernelType($node)) {
+            return [];
+        }
```

Comment:
This seems like a lot of code and local logic.

Could you extract it to own service in `src/NodeAnalyzer` namespace?

Reply:
I will try

Reply:
I tried add `DependencyNodeAnalyzer` inside `Symplify\CodingStandard\NodeAnalyzer` namespace under `packages/coding-standard/src/NodeAnalyzer` directory, and inject into `ForbiddenProtectedPropertyRule` constructor, and got the following error:

```bash
Service of type Symplify\CodingStandard\Rules\ForbiddenProtectedPropertyRule: Service of type Symplify\CodingStandard\NodeAnalyzer\DependencyNodeAnalyzer 

needed by $dependencyNodeAnalyzer in __construct() not found. 

Did you add it to configuration file?
```

Do you have information how to register the new class as a service ?

Reply:
I found it, it can be registered to `packages/coding-standard/config/symplify-rules.neon`

Reply:
implemented.